tests/Backend/Workspaces/WorkspacesTest.php

```diff
@@ -77,9 +77,21 @@ public function testWorkspacePasswordReset()
 
         $this->assertArrayHasKey("mytable", array_flip($tableNames));
 
+        $runId = $this->_client->generateRunId();
+        $this->_client->setRunId($runId);
+
         $newCredentials = $workspaces->resetWorkspacePassword($workspace['id']);
         $this->assertArrayHasKey("password", $newCredentials);
 
+        // block until async events are processed, processing in order is not guaranteed but it should work most of time
+        $this->createAndWaitForEvent((new \Keboola\StorageApi\Event())->setComponent('dummy')->setMessage('dummy'));
+
+        $events = $this->_client->listEvents([
+            'runId' => $runId,
+        ]);
+
+        var_export($events);
```

Comment:
Tady se mi to nějak nedaří (podobně jako u workspace create a delete - eventa se teda vytvoří, ale nemá přiřazené `runId`, nevíš proč @Halama

Reply:
takže ti jí to nevrátí v tech eventách vyfiltrovaných podle run id?

Reply:
už to vidím ono se tam pro tu synchronní akci narozdíl od async jobu nepředává automaticky na backendu:(

Reply:
Asi nechápu, jak to myslíš. Koukal jsem třeba u LoadClone eventy a tam se to do ni nijak taky nedává... resp. našel jsem, že se to do ní doplňuje v `\Service_EventDispatcher::_logAppEvent`. Ale jen pokud je to instance `Event_Aplication`. Nicméně to by měly být všechny.

Reply:
Jo jasně - jenže do toho dispatcheru se to sype jen v ExecuteJob a nikde jinde. No a je otázka, jestli by i synchronní volání měly mít runId. Asi k tomu není úplně důvod, kromě toho aby se to dalo tahat v tom testu.

Reply:
Na to, že tam nebudou padat jiné eventy asi moc spoléhat nemůžu, když se to pak spouští paralelně ve více vláknech :(

Reply:
jo měly by mít runId když ho dostanou. Např. synchronní náhledy tabulek apod. se s runId párují.